src/services/symbolDisplay.ts

```diff
@@ -296,7 +296,7 @@ namespace ts.SymbolDisplay {
             }
             else {
                 // Method/function type parameter
-                let declaration = <Node>getDeclarationOfKind(symbol, SyntaxKind.TypeParameter);
+                let declaration: Node = getDeclarationOfKind(symbol, SyntaxKind.TypeParameter);

```

Comment:
The problem is that it's typed with `<T extends Declaration>`. The `: Node` was just there so that `declaration` could be reassigned later. Fixed by using two separate `const` declarations instead of the `let`.